# Customer Personality Analysis for Marketing Campaign
Exploratory Data Analysis Excercise conducted by **Widya Ayuningtyas**


>  Objective

* Which product purchased the most by group of customers?
* Which places are getting the most purchase?

> Restriction

* Not using data in category promotion as it is irrelevant to objective

> Methodology
* Descriptive Analysis
* Sum Analysis

> Included Packages
* Pandas
* Numpy

> Expected Outcome
* Most purchased products by category
* Most visited store

> Steps are as following :


## **Data Understanding**

Source : The dataset for this project is provided by Dr. Omar Romero-Hernandez.
https://www.kaggle.com/imakash3011/customer-personality-analysis

Marketing Campaign Data has 29 columns each are from 4 category below :

1. **People**
* ID: Customer's unique identifierYear_Birth: Customer's birth year
* Education: Customer's education level
* Marital_Status: Customer's marital status
* Income: Customer's yearly household income
* Kidhome: Number of children in customer's household
* Teenhome: Number of teenagers in customer's household
* Dt_Customer: Date of customer's enrollment with the company
* Recency: Number of days since customer's last purchase
* Complain: 1 if the customer complained in the last 2 years, 0 otherwise
2. **Products**
* MntWines: Amount spent on wine in last 2 years
* MntFruits: Amount spent on fruits in last 2 years
* MntMeatProducts: Amount spent on meat in last 2 years
* MntFishProducts: Amount spent on fish in last 2 years
* MntSweetProducts: Amount spent on sweets in last 2 years
* MntGoldProds: Amount spent on gold in last 2 years
3. **Purchase**
* NumDealsPurchases: Number of purchases made with a discount
* NumWebPurchases: Number of purchases made through the company’s web site
* NumCatalogPurchases: Number of purchases made using a catalogue
* NumStorePurchases: Number of purchases made directly in stores
* NumWebVisitsMonth	: Number of visitation on website in month
4. **Promotion**
* AcceptedCmp1: 1 if customer accepted the offer in the 1st campaign, 0 otherwise
* AcceptedCmp2: 1 if customer accepted the offer in the 2nd campaign, 0 otherwise
* AcceptedCmp3: 1 if customer accepted the offer in the 3rd campaign, 0 otherwise
* AcceptedCmp4: 1 if customer accepted the offer in the 4th campaign, 0 otherwise
* AcceptedCmp5: 1 if customer accepted the offer in the 5th campaign, 0 otherwise
* Response: 1 if customer accepted the offer in the last campaign, 0 otherwise
4. **Place**
* NumWebPurchases: Number of purchases made through the company’s website
* NumCatalogPurchases: Number of purchases made using a catalogue
* NumStorePurchases: Number of purchases made directly in stores
* NumWebVisitsMonth: Number of visits to company’s website in the last month

## Data Preparation





### Installing Packages

In [1]:
# Standard Library
import pandas as pd
import numpy as np


### Importing Data

In [42]:
data = pd.read_csv('/content/marketing_campaign.csv', sep="\t")
data.head()

ID  Year_Birth   Education  ... Z_CostContact  Z_Revenue  Response
0  5524        1957  Graduation  ...             3         11         1
1  2174        1954  Graduation  ...             3         11         0
2  4141        1965  Graduation  ...             3         11         0
3  6182        1984  Graduation  ...             3         11         0
4  5324        1981         PhD  ...             3         11         0

[5 rows x 29 columns]

### Analyze Data

In [43]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

In [44]:
(24/2240)*100

1.0714285714285714

There are 24 missing data from Income. It's about 1.07% of total data, as we can't confirming nor collect the missing data, we can drop 24 missing data.

In [60]:
#Check missing data
data.isnull().sum()

ID                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Complain                0
Z_CostContact           0
Z_Revenue               0
Response                0
dtype: int64

In [55]:
df = data.dropna()

> Dropping columns from all promotion, as those will not be used in the analysis;
> Dropping column Z_CostContact and Z_Revenue as these two columns have same value in all data points. So these two columns are not relevant and can be dropped.

In [56]:
df = df.drop(['AcceptedCmp3','AcceptedCmp4','AcceptedCmp5','AcceptedCmp1','AcceptedCmp2','Response','Z_Revenue', 'Z_CostContact'], axis = 1)
df.head()

ID  Year_Birth   Education  ... NumStorePurchases  NumWebVisitsMonth  Complain
0  5524        1957  Graduation  ...                 4                  7         0
1  2174        1954  Graduation  ...                 2                  5         0
2  4141        1965  Graduation  ...                10                  4         0
3  6182        1984  Graduation  ...                 4                  6         0
4  5324        1981         PhD  ...                 6                  5         0

[5 rows x 21 columns]

In [57]:
df.shape

(2216, 21)

In [58]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2216 entries, 0 to 2239
Data columns (total 21 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2216 non-null   int64  
 1   Year_Birth           2216 non-null   int64  
 2   Education            2216 non-null   object 
 3   Marital_Status       2216 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2216 non-null   int64  
 6   Teenhome             2216 non-null   int64  
 7   Dt_Customer          2216 non-null   object 
 8   Recency              2216 non-null   int64  
 9   MntWines             2216 non-null   int64  
 10  MntFruits            2216 non-null   int64  
 11  MntMeatProducts      2216 non-null   int64  
 12  MntFishProducts      2216 non-null   int64  
 13  MntSweetProducts     2216 non-null   int64  
 14  MntGoldProds         2216 non-null   int64  
 15  NumDealsPurchases    2216 non-null   i

In [59]:
df.describe()

ID   Year_Birth  ...  NumWebVisitsMonth     Complain
count   2216.000000  2216.000000  ...        2216.000000  2216.000000
mean    5588.353339  1968.820397  ...           5.319043     0.009477
std     3249.376275    11.985554  ...           2.425359     0.096907
min        0.000000  1893.000000  ...           0.000000     0.000000
25%     2814.750000  1959.000000  ...           3.000000     0.000000
50%     5458.500000  1970.000000  ...           6.000000     0.000000
75%     8421.750000  1977.000000  ...           7.000000     0.000000
max    11191.000000  1996.000000  ...          20.000000     1.000000

[8 rows x 18 columns]

### Categorizing Customer by Age and Income

In [69]:
# add customer categorization by income & age
df['age'] = 2022 - df['Year_Birth']

df['AgeRange'] = pd.cut(
                        data['age'],
                        bins=[0,13,21,40,90],
                        labels=['Child','Teens','Adult','Old'])

df['IncomeType'] = pd.cut(
                          data['Income'],
                          bins=[0,35000,50000,70000,666666],
                          labels=['Low','Medium','High','Very High'])
df.head()

ID  Year_Birth   Education  ... age  IncomeType  AgeRange
0  5524        1957  Graduation  ...  65        High       Old
1  2174        1954  Graduation  ...  68      Medium       Old
2  4141        1965  Graduation  ...  57   Very High       Old
3  6182        1984  Graduation  ...  38         Low     Adult
4  5324        1981         PhD  ...  41        High       Old

[5 rows x 24 columns]

In [ ]:
,

### Which Product are purchased the most by Customer Age and Income

In [90]:
df1 = df.groupby(["AgeRange"])["MntWines","MntFruits","MntMeatProducts","MntFishProducts","MntSweetProducts","MntGoldProds"].sum().reset_index()
df1.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


AgeRange  MntWines  ...  MntSweetProducts  MntGoldProds
0    Child         0  ...                 0             0
1    Teens         0  ...                 0             0
2    Adult     87466  ...              9979         14892
3      Old    587839  ...             49849         82284

[4 rows x 7 columns]

By age range, the most purchased product by :

*   Adult : Wines
*   Old : Meat



In [94]:
df2 = df.groupby(["IncomeType"])["MntWines","MntFruits","MntMeatProducts","MntFishProducts","MntSweetProducts","MntGoldProds"].sum().reset_index()
df2.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


IncomeType  MntWines  ...  MntSweetProducts  MntGoldProds
0        Low     11707  ...              3198          8915
1     Medium     59998  ...              4225         13990
2       High    278701  ...             18500         37310
3  Very High    325677  ...             33973         37212

[4 rows x 7 columns]

By income type, the most purchased product are :

*   Low : Meat
*   Medium : Wines
*   High : Wines
*   Very High : Wines



### Which places are getting the most purchase

In [97]:
df3 = df.groupby(["AgeRange"])["NumWebPurchases","NumCatalogPurchases","NumStorePurchases"].sum().reset_index()
df3.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


AgeRange  NumWebPurchases  NumCatalogPurchases  NumStorePurchases
0    Child                0                    0                  0
1    Teens                0                    0                  0
2    Adult             1134                  850               1805
3      Old             7912                 5062              11042

By age range, the place getting the most purchase are : 

*   Adult : Store
*   Old : Store



In [96]:
df4 = df.groupby(["IncomeType"])["NumWebPurchases","NumCatalogPurchases","NumStorePurchases"].sum().reset_index()
df4.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


IncomeType  NumWebPurchases  NumCatalogPurchases  NumStorePurchases
0        Low             1098                  259               1623
1     Medium             1681                  572               2101
2       High             3605                 2131               4913
3  Very High             2669                 2957               4218

By income type, the place getting the most purchase are :

*   Low : Store
*   Medium : Store
*   High : Store
*   Very High : Store



## Recommendation

1. Maximizing promoting wines for adult category and higher income, then promoting meat for old category and lower income
2. Maximizing sales and promoting more in store as most purchases are coming from store.

